core/src/main/scala/zio/logging/Logger.scala

```diff
@@ -0,0 +1,64 @@
+package zio.logging
+
+import zio.clock._
+import zio.console._
+import zio.internal.Tracing
+import zio.internal.stacktracer.Tracer
+import zio.internal.stacktracer.ZTraceElement.{ NoLocation, SourceLocation }
+import zio.internal.stacktracer.impl.AkkaLineNumbersTracer
+import zio.internal.tracing.TracingConfig
+import zio.{ FiberRef, UIO, ZIO }
+
+/**
+ * A logger of strings.
+ */
+trait Logger extends LoggerLike[String]
+object Logger {
+
+  private val tracing = Tracing(Tracer.globallyCached(new AkkaLineNumbersTracer), TracingConfig.enabled)
+
+  private[logging] def classNameForLambda(lambda: => AnyRef) =
+    tracing.tracer.traceLocation(() => lambda) match {
+      case SourceLocation(_, clazz, _, _) => Some(clazz)
+      case NoLocation(_)                  => None
+    }
+
+  /**
+   * Constructs a logger provided the specified sink.
+   */
+  def make(logLevel: LogLevel, logger: (LogContext, => String) => UIO[Unit]): UIO[Logger] =
```

Comment:
Generalize `make` to `def make[R](logLevel: LogLevel, logger: (LogContext, => String) => URIO[R, Any]): URIO[R, Logger]`. 

This will clean up some code down below.